controllers/operatorconfig_controller.go

```diff
@@ -54,51 +78,324 @@ type OperatorConfigReconciler struct {
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: InstallNamespace,
+		Name:      OwnerConfigName,
+	}, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("Desired OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Add Finalizer for this CR
+	if !controllerutil.ContainsFinalizer(operatorConfig, Finalizer) {
+		controllerutil.AddFinalizer(operatorConfig, Finalizer)
+		err := r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if operatorConfig.DeletionTimestamp != nil {
+		// clean up
+		err := r.CleanUpResources(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		controllerutil.RemoveFinalizer(operatorConfig, Finalizer)
+		err = r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		log.Info("Clean up successful")
+		return ctrl.Result{}, nil
+	}
 
-	err := createClusterTasks(ctx, log)
+	for _, f := range filepaths {
+		err := r.createResourcesFromFile(f, ctx, log, operatorConfig)
+		if err != nil {
+			log.Error(err, "Error creating resources")
+			return ctrl.Result{Requeue: true}, nil
+		}
+	}
+
+	log.Info("All needed resources are present")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorConfigReconciler) CleanUpResources(ctx context.Context) error {
+
+	for _, f := range filepaths {
+		err := r.deleteResourcesFromFile(f, ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
 
+func (r *OperatorConfigReconciler) createResourcesFromFile(path string, ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
+
+	var decoder = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+	data, err := getResources(path)
 	if err != nil {
-		log.Error(err, "Error creating cluster tasks")
-		return ctrl.Result{Requeue: true}, nil
+		return err
+	}
+
+	reconcilersForGVK := map[string]func(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error{
+		"Deployment":    r.reconcileDeployment,
+		"Service":       r.reconcileService,
+		"ConfigMap":     r.reconcileConfigMap,
+		"ClusterTask":   r.reconcileClusterTask,
+		"ConsolePlugin": r.reconcileConsolePlugin,
+	}
+
+	for _, resource := range data {
+		if len(resource) > 0 {
+			obj := unstructured.Unstructured{}
+			_, gvk, err := decoder.Decode([]byte(resource), nil, &obj)
+			if err != nil {
+				return err
+			}
+
+			if fn, ok := reconcilersForGVK[gvk.Kind]; ok {
```

Comment:
nit: change `fn` to `reconcile`? sorry I realize I suggested it but could have done better at naming